addon-test-support/-private/setup-window-mock.js

```diff
@@ -1,11 +1,15 @@
-import { reset } from 'ember-window-mock';
+import { reset, mockProxyHandler } from './window';
+import { _setCurrentHandler } from 'ember-window-mock';
 
 //
 // Used to reset window-mock for a test.
 //
 // NOTE: the `hooks = self` is for mocha support
 //
 export default function setupWindowMock(hooks = self) {
-  // @TODO: assert that window mock is pristine in `hooks.before`.
-  hooks.afterEach(reset);
+  hooks.beforeEach(() => _setCurrentHandler(mockProxyHandler))
+  hooks.afterEach(() => {
+    reset();
+    _setCurrentHandler(null);
```

Comment:
@simonihmig - Sorry for the late comment, I'm curious if you think this should be a stack instead of always resetting to `null`?

Specifically, should we reset to `null` or to whatever the value was _before_? What happens if you do:

```js
module('foo', function(hooks) {
  setWindowMock(hooks);

  module('nested thing', function(hooks) {
    setWindowMock(hooks);
  });

  test('something that requires the window mock', function() {
    // ...snip...
  });
});
```

Reply:
It is also vaguely weird that it would reset with `null`. I would think that you would want to call `_setCurrentHandler()` which would reset to `defaultHandler` (AFAICT calling with `null` will make any `window` access fail).

Reply:
@rwjblue thanks for the review and the suggestions, I think they make totally sense! Especially that `null` thing was stupid, apparently I thought this would reset to `defaultHandler`, which you are right it does *not*. I should add some better test coverage, for what state `setupWindowMock` leaves behind afterwards, especially for that nested usage...

Reply:
For the record, as of `0.7.1` it no longer resets to `null`; it now returns to the default handler after each test.